docs/DeveloperGuide.md

```diff
@@ -6,33 +6,230 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### The Overall UML Diagram:
 
+![UML Diagram](assets/images/mathGenius.png)    
+
+
+### The overall Sequence Diagram:
+![Sequence Diagram](assets/images/sequenceDiagram.png)    
```

Comment:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/68402849/4c68dca3-fdda-4ad1-ad39-01f6e88e6a4b)
Could an action bar be missing for writeFile()